ignite/cmd/plugin.go

```diff
@@ -376,18 +382,100 @@ plugins:
 	}
 }
 
-func printPlugins() {
-	if len(plugins) == 0 {
-		fmt.Println("No plugin found")
-		return
+func NewPluginDescribe() *cobra.Command {
+	return &cobra.Command{
+		Use:   "describe [path]",
+		Short: "outputs information about the specified plugin. must be registered",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			s := cliui.New(cliui.WithStdout(os.Stdout))
+
+			for _, p := range plugins {
+				if p.Path == args[0] {
+					manifest, err := p.Interface.Manifest()
+					if err != nil {
+						return fmt.Errorf("error while loading plugin manifest\n%s", err)
+					}
+
+					printPluginCommands(manifest.Commands, s)
+					printPluginHooks(manifest.Hooks, s)
+					break
+				}
+			}
+
+			return nil
+		},
 	}
+}
+
+func printPlugins(session *cliui.Session) error {
 	var entries [][]string
 	for _, p := range plugins {
 		status := "✅ Loaded"
 		if p.Error != nil {
 			status = fmt.Sprintf("❌ Error: %v", p.Error)
 		}
+
+		manifest, err := p.Interface.Manifest()
+		if err != nil {
+			return fmt.Errorf("error while loading plugin manifest\n%s", err)
+		}
+
+		var (
+			hookCount = len(manifest.Hooks)
+			cmdCount  = len(manifest.Commands)
+		)
+
+		status = fmt.Sprintf("%s 🪝 %d 💻 %d", status, hookCount, cmdCount)
 		entries = append(entries, []string{p.Path, status})
 	}
-	entrywriter.MustWrite(os.Stdout, []string{"path", "status"}, entries...)
+
+	session.PrintTable([]string{"Path", "Status"}, entries...)
+
+	return nil
+}
+
+func printPluginCommands(cmds []plugin.Command, session *cliui.Session) {
+	var entries [][]string
+	// Processes command graph
+	traverse := func(cmd plugin.Command) {
+		// cmdPair is a Wrapper struct to create parent child relationship for sub commands without a `place command under`
+		type cmdPair struct {
+			cmd    plugin.Command
+			parent plugin.Command
+		}
+
+		queue := make([]cmdPair, 0)
+		queue = append(queue, cmdPair{cmd: cmd, parent: plugin.Command{}})
+
+		for len(queue) > 0 {
+			c := queue[0]
+			queue = queue[1:]
+			if c.cmd.PlaceCommandUnder != "" {
+				entries = append(entries, []string{c.cmd.Use, c.cmd.PlaceCommandUnder})
+			} else {
+				entries = append(entries, []string{c.cmd.Use, c.parent.Use})
+			}
+
+			for _, sc := range c.cmd.Commands {
+				queue = append(queue, cmdPair{cmd: sc, parent: c.cmd})
+			}
+		}
+	}
+
+	for _, c := range cmds {
+		traverse(c)
+	}
+
+	session.PrintTable([]string{"command use", "under"}, entries...)
```

Comment:
Let's say I have a plugin that adds 2 commands, each of this command has 3 sub-commands, the output of `describe` will look like :

```
Command Use 	Under 		
my-new-plugin 	ignite 		
add 		my-new-plugin 	
list 		my-new-plugin 	
delete 		my-new-plugin 	
my-new-plugin2 	scaffold 	
add 		my-new-plugin2 	
list 		my-new-plugin2 	
delete 		my-new-plugin2 	
```

I find this a little confusing and verbose. I think we should reduce/simplify the output size, by removing the sub-commands (we don't really care at this point, user will discover them by running the plugin command), and prints the full command directly. Something like :

```
Command
ignite my-new-plugin
ignite scaffold my-new-plugin2 
```

Note that `ignite` prefix is added if absent, like already done in other plugin related code.

This should simplify a lot the ` printPluginCommands` func.

Reply:
I can see how the current out put can be confusing. But I think being able to see nested commands is useful for larger plugins. Could we only traverse the command graph if a flag is used like '--verbose'. Also instead of just printing the command it's directly under it will print the full command path.

Reply:
OK, fine for me :+1: